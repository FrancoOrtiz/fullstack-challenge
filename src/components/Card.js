import React from 'react'

export const Card = () => {
  return (
    <div className="flex w-80 h-40 justify-between my-10">
      <div className="h-full w-40 bg-gray-500">
        <img src=""></img>
      </div>
      <span className="mt-2 text-xl text-gray-500">Pikachu</span>
    </div>
  )
}
