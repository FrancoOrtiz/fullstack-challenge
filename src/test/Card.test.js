/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { Card } from "../components/Card.js";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";

describe("Tests on Card.js", () => {
  let wrapper;

  const Info = {
    pokemonName: "pikachu",
    pokemonImg:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
    pokemonNum: 35,
  };

  const { pokemonName, pokemonImg, pokemonNum } = Info;

  test("should be render", () => {
    wrapper = renderer
      .create(
        <Card
          pokemonName={pokemonName}
          pokemonImg={pokemonImg}
          pokemonNum={pokemonNum}
        />
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  test("should be display correctly", () => {
    wrapper = render(
      <Card
        pokemonName={pokemonName}
        pokemonImg={pokemonImg}
        pokemonNum={pokemonNum}
      />
    );

    wrapper.getByText("pikachu");
    wrapper.getByText(35);

    const img = wrapper.getByRole("img").src;

    expect(img).toEqual(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
    );
  });
});
