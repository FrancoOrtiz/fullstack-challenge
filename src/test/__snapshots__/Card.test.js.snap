// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on Card.js should be render 1`] = `
<div
  className="flex w-80 h-40 my-10"
>
  <div
    className="h-full w-40 bg-gray-100 border-4 border-blue-300 rounded-full"
  >
    <img
      alt=""
      className="w-full"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
    />
  </div>
  <div
    className="mt-4 ml-10 flex flex-col"
  >
    <span
      className="text-xl font-bold text-gray-400"
    >
      pikachu
    </span>
    <span>
      35
    </span>
  </div>
</div>
`;
